Common/Models/StatusExtensionContext.swift

```diff
@@ -29,6 +29,12 @@ struct NetBasalContext {
     let end: Date
 }
 
+struct ActiveInsulinContext {
+    let value: Double
+    let startDate: Date
+    let error: String
```

Comment:
I'm not sure that it makes sense to pass the actual error string through to this point. showing it in the widget likely won't make a lot of sense. I think you have a few options:

1) if there's an error, don't add an ActiveInsulinContext and leave it blank in the widget
2) if there's an error, don't add an ActiveInsulinContext and report "ERR" in the widget
3) let error be an optional String so that you're not checking for "", but for the existence of the error